src/stream_wrap.cc

```diff
@@ -272,14 +271,14 @@ void LibuvStreamWrap::OnUvRead(ssize_t nread, const uv_buf_t* buf) {
           object()->Set(env()->context(),
                         env()->pending_handle_string(),
                         local_pending_obj).IsNothing()) {
-      return;
+      return v8::Nothing<void>();
```

Comment:
If the `ToLocal()` call fails and return `false` it would indicate that there is a pending exception, so we should return `Nothing<void>()` immediately from this function but the current implementation would proceed to run `EmitRead()` and then return `JustVoid()` which would indicate that no exception was thrown, so this is not correct. We should write it like this:
```cc
if (!pending_obj.ToLocal(&local_pending_obj) ||
       object()->Set(env()->context(),
                     env()->pending_handle_string(),
                     local_pending_obj).IsNothing()) {
  return Nothing<void>();
}
```

Reply:
Your interpretation looks correct. However, it seems to be required to run `EmitRead()`  irrelevant to the existence of `pending_obj`. Without it, many tests fail.

If my observation is correct, could I know which one below you think is reasonable?

- Reverting to returning `void` if we could say that the function can handle a pending exception by itself.
- Or else, returning `Nothing<void>` after `EmitRead()` in case that `pending_obj.ToLocal()` returned `false`.

```c++
  bool has_no_pending_obj = true;
 
  ....

  if (bread > 0) {
    Local<Object> local_pending_obj;
    has_no_pending_obj = pending_obj.ToLocal(&local_pending_obj);
    if (has_no_pending_obj &&
          object()->Set(env()->context(),
                        env()->pending_handle_string(),
                        local_pending_obj).IsNothing()) {
      return Nothing<void>();
    }
  }

  EmitRead(nread, *buf);
  return has_no_pending_obj ? JustVoid() : Nothing<void>();
}
```

Reply:
@dayeon thanks for checking! I found the reason why that doesn't work - if `type` is set to `UV_UNKNOWN_HANDLE`, the `pending_obj` Maybe isn't assigned anything, so `IsNothing()` would return `true` although no exception is being thrown. Additionally checking `type` in the conditional should fix it: (I just ran the `stream`-related tests with `python3 tools/test.py --mode=release stream -j 16` and all of those are passing with this patch but I haven't run the linter, so you might want to do that before pushing)
```cc
    if (type != UV_UNKNOWN_HANDLE &&
          (!pending_obj.ToLocal(&local_pending_obj) ||
          object()->Set(env()->context(),
                        env()->pending_handle_string(),
                        local_pending_obj).IsNothing())) {
      return Nothing<void>();
    }
```
Feel free to split the conditional into multiple ones if you feel that it's more readable that way. :)

Reply:
Wow, thanks for the clarification! Applied the fix.